### PR TITLE
Alton/auth api

### PR DESCRIPTION
**Pull request by altonzheng**
- Added user model
- Created sessions controller for logging in/logging out
- Generate auth tokens and check for every API call
- Search and create salesforce data (TODO: actually post to salesforce)
- Fix rake task for pulling accounts from Salesforce (TODO: Fix sql injection vulnerability/refactor)

### CODE REVIEW

**Comment by altonzheng** [comment]
@SamLau95 @chenhoward Can you guys take a look? Thanks!

File: phc-rails/Rakefile [hunk 1]
```diff
@@ -19,13 +19,18 @@ task :load_sf => :environment do
   order_str = "ORDER BY LastModifiedDate ASC Limit 100"
   field_str = "SELECT Id, FirstName, LastName, PersonBirthdate, LastModifiedDate from Account "
   username = ENV["sf_username"]
-  password = ENV["sf_password"]
+  password = ENV["sf_password"] + ENV["sf_security_token"]
+
+  byebug
```
Comment (chenhoward) [inline phc-rails/Rakefile]: What is this?

File: phc-rails/app/assets/javascripts/sessions.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+// Place all the behaviors and hooks related to the matching controller here.
```
Comment (chenhoward) [inline phc-rails/app/assets/javascripts/sessions.js]: I would take out these empty files

File: phc-rails/app/controllers/api/v1/accounts_controller.rb [hunk 3]
```diff
@@ -1,31 +1,55 @@
...
-      lhash = { last_name: last_name }
-      res = PersonAccount.fuzzy_search(fhash, false).fuzzy_search(lhash, false)
-      respond_with res
+    auth_token = request.headers["HTTP_AUTH_TOKEN"]
+    user_id = request.headers["HTTP_USER_ID"]
+    if auth_token && user_id
+      if user_authenticated?(user_id, auth_token)
+        first_name = request.headers["HTTP_FIRSTNAME"]
+        last_name = request.headers["HTTP_LASTNAME"]
+        result = PersonAccount.fuzzy_search({first_name: first_name}, false).fuzzy_search({last_name: last_name}, false)
```
Comment (chenhoward) [inline phc-rails/app/controllers/api/v1/accounts_controller.rb]: I would actually change what I did here from fuzzy searching in the controller to doing most of the logic in the model.

File: phc-rails/app/controllers/api/v1/accounts_controller.rb [hunk 4]
```diff
@@ -1,31 +1,55 @@
...
+    auth_token = request.headers["HTTP_AUTH_TOKEN"]
+    user_id = request.headers["HTTP_USER_ID"]
+    if auth_token && user_id
+      if user_authenticated?(user_id, auth_token)
+        person = PersonAccount.new
+        person.first_name = params[:first_name]
+        person.last_name = params[:last_name]
+        person.sf_id = request.headers[:salesforceid]
+        person.save
+        respond_with "Successfully saved user!", status: 200, :location => root_url
```
Comment (chenhoward) [inline phc-rails/app/controllers/api/v1/accounts_controller.rb]: Agreed with this

File: phc-rails/app/controllers/api/v1/accounts_controller.rb [hunk 5]
```diff
@@ -1,31 +1,55 @@
...
+
+        ##### TODO: POST TO SALESFORCE #####
+      end
+    else
+      respond_with "Error: Please include authentication token.", status: 401, location => root_ur
```
Comment (chenhoward) [inline phc-rails/app/controllers/api/v1/accounts_controller.rb]: I would do hash syntax here too

File: phc-rails/app/helpers/users_helper.rb [hunk 6]
```diff
@@ -0,0 +1,2 @@
+module UsersHelper
```
Comment (chenhoward) [inline phc-rails/app/helpers/users_helper.rb]: Can take out this empty thing

File: phc-rails/app/controllers/users_controller.rb [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+  def show
+    @user = User.find(params[:id])
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def index
```
Comment (chenhoward) [inline phc-rails/app/controllers/users_controller.rb]: Gotta make sure eventaully only admins can see all users

File: phc-rails/config/application.yml.sample [hunk 8]
```diff
@@ -0,0 +1,12 @@
+# Add configuration values here, as shown below.
```
Comment (chenhoward) [inline phc-rails/config/application.yml.sample]: Lol I freak out every time I see this

**Comment by twuster** [comment]
LGTM

**Comment by SamLau95** [comment]
Yo wat none of the comments got addressed

**Comment by varunrau** [comment]
LOL hound --

**Comment by jjwon** [comment]
tdawg says YOLOOOO

**Comment by SamLau95** [comment]
no guys you can't just do this

**Pull request merged**